R/export.R

```diff
@@ -143,29 +143,26 @@ make_ret_type <- function(parsed, type = c("unchanged", "c_style", "rcpp")) {
 make_c_types <- function(types) {
   types <- ifelse(types == "torch::Tensor", "void*", types)
   types <- ifelse(types == "std::vector<torch::Tensor>", "void*", types)
+  types <- ifelse(types == "torch::optional<torch::Tensor>", "void*", types)
   types
 }
 
 make_lantern_type <- function(types) {
   types <- ifelse(types == "torch::Tensor", "Tensor", types)
   types <- ifelse(types == "std::vector<torch::Tensor>", "TensorList", types)
+  types <- ifelse(types == "torch::optional<torch::Tensor>", "OptionalTensorList", types)
```

Comment:
You want this to call: https://github.com/mlverse/torch/blob/master/lantern/include/lantern/types.h#L105 or https://github.com/mlverse/torch/blob/master/lantern/include/lantern/types.h#L120 when interpolated. So I think you want:

```suggestion
  types <- ifelse(types == "torch::optional<torch::Tensor>", "optional::Tensor", types)
```